x-pack/legacy/plugins/epm/server/datasources/create.ts

```diff
@@ -78,12 +78,12 @@ async function saveDatasourceReferences(options: {
   };
 
   const toInstall = (toSave as Asset[]).reduce(assetsReducer, savedAssets);
+
+  // Works, but would still like to improve the DX here
   const datasource: Datasource = createFakeDatasource(pkg, toInstall);
-  // ideally we'd call .create from /x-pack/legacy/plugins/ingest/server/libs/datasources.ts#L22
-  // or something similar, but it's a class not an object so many pieces are missing
-  // we'd still need `user` from the request object, but that's not terrible
-  // lacking that we make another http request to Ingest
-  await ingestDatasourceCreate({ request, datasource });
+  const ingestPlugin = request.server.plugins.ingest;
+  const user = HapiFrameworkAdapter.getUserFromRequest(request);
+  await ingestPlugin.datasources.create(user, datasource);
```

Comment:
How bad is it to keep making the extra request?  I feel like we should try to use the http endpoint if we cant, but I'm not sure how common this scenario is. What's the major downside?  Having to pass the request down?